static/src/javascripts/projects/commercial/modules/dfp/render-advert.ts

```diff
@@ -39,16 +40,16 @@ const addClassIfHasClass = (newClassNames: string[]) =>
 					}).get('isDotcomRendering', false);
 					if (
 						isDotcomRendering &&
+						!allClasses.contains('ad-slot--carrot') &&
 						newClassNames.includes('ad-slot--fluid')
```

Comment:
Could this now be changed to use `allClasses` too?

Reply:
Actually, this should only apply if you’re trying to add the `ad-slot--fluid` class, so no. But we should extract this css logic and add it to https://github.com/guardian/dotcom-rendering/blob/main/src/web/components/ArticleContainer.tsx

Reply:
Okay. It just seemed because of [this line](https://github.com/guardian/frontend/blob/494c577fe48955e1745fc7e1bd80010b01b75951/static/src/javascripts/projects/commercial/modules/dfp/render-advert.ts#L31) it looks like `allClasses` is a superset of `newClassNames` so `contains` should work out the same.

Reply:
True, but that might mean that you’re applying those styles many times over. In any case, I’m looking at removing JS-injected styles for DCR and doing it properly!